src/app/static/src/tests/components/userHeader.test.ts

```diff
@@ -108,4 +124,108 @@ describe("user header", () => {
         reader.readAsText(actualBlob);
     });
 
+    it ("opens file dialog on click load", (done) => {
+        const wrapper = shallowMount(UserHeader,
+            {
+                store: createStore()
+            });
+
+        wrapper.find(".dropdown-toggle").trigger("click");
+        expect(wrapper.find(".dropdown-menu").classes()).toStrictEqual(["dropdown-menu", "show"]);
+        const link = wrapper.findAll(".dropdown-item").at(1);
+
+        const input = wrapper.find("input").element as HTMLInputElement;
+        input.addEventListener("click", function(){
+            //file dialog was opened
+            done();
+        });
+
+        link.trigger("mousedown");
+    });
+
+    it("invokes load action when file selected from dialog", () => {
+        const mockLoadAction = jest.fn();
+        const wrapper = shallowMount(UserHeader,
+            {
+                store: new Vuex.Store({
+                    modules: {
+                        baseline: {
+                            namespaced: true,
+                            state: mockBaselineState()
+                        },
+                        surveyAndProgram: {
+                            namespaced: true,
+                            state: mockSurveyAndProgramState()
+                        },
+                        modelRun: {
+                            namespaced: true,
+                            state: mockModelRunState()
+                        },
+                        load: {
+                            namespaced: true,
+                            state: mockLoadState(),
+                            actions: {
+                                load: mockLoadAction
+                            }
+                        }
+                    }
+                })
+            });
+
+        const input = wrapper.find("input");
+
+        const vm = wrapper.vm;
+        const testFile = new File(["test file contents"], "filename");
+        //Can't programmatically construct a FileList to give to the read rendered input element, so we need to trick
+        //the component with a mocked ref
+        (vm.$refs as any).loadFile = {
+            files: [testFile]
+        };
+
+        input.trigger("change");
+        expect(mockLoadAction.mock.calls.length).toEqual(1);
+        expect(mockLoadAction.mock.calls[0][1]).toBe(testFile)
+    });
+
+    it("does not open modal if no load error", () => {
+        const wrapper = shallowMount(UserHeader,
+            {
+                store: createStore()
+            });
+
+        expect(wrapper.find(Modal).attributes("open")).toBeFalsy();
+    });
+
+    it("opens modal if load error", () => {
+        const wrapper = shallowMount(UserHeader,
```

Comment:
Can pull the wrapper creation out into a shared function to be re-used between this and the test above?

Reply:
There's already a createStore method, I'll use that and just poke in the load module bits I need, should be much terser.